components/mediation/registry/org.wso2.micro.integrator.registry/src/main/java/org/wso2/micro/integrator/registry/MicroIntegratorRegistry.java

```diff
@@ -115,6 +121,34 @@ public MicroIntegratorRegistry() {
         this.govRegistry = getUri(defaultFSRegRoot, "governance");
 
         initSecurityRepo();
+        initRegistryListener(defaultFSRegRoot);
+    }
+
+    private void initRegistryListener(String regRoot) {
+        try {
+            MicroIntegratorRegistryListener watcher = new MicroIntegratorRegistryListener(Paths.get(regRoot));
+            ExecutorService executor = Executors.newSingleThreadExecutor();
+            executor.submit(watcher);
+            executor.shutdown();
```

Comment:
Is it correct to shutdown the executor here?

Reply:
Yes, this is to indicate that there is no task anymore to submit to the executor